src/main/java/com/agnieszkapawska/flashcards/domain/repositories/FlashcardRepository.java

```diff
@@ -2,10 +2,11 @@
 
 import com.agnieszkapawska.flashcards.domain.models.Flashcard;
 import org.springframework.data.jpa.repository.JpaRepository;
-
+import java.util.List;
 import java.util.Optional;
 
 public interface FlashcardRepository extends JpaRepository<Flashcard, Long> {
 
-    Optional<Flashcard> findByQuestion(String question);
+    Optional<List<Flashcard>> findByQuestionContaining(String searchPhrase);
+    Optional<List<Flashcard>> findByAnswerContaining(String search);
```

Comment:
- search -> searchPhrase
- IgnoreCaseContaining
- have you tried to join those methods into one? https://stackoverflow.com/a/47785752